businessRules/users.js

```diff
@@ -4,61 +4,72 @@ const { createUser } = require('../models/users');
 const { getUsers, setUsers } = require('../utils/storage');
 
 function registerUser(userName, password, firstName, lastName) {
+    if (!userName) {
+        throw new Error('Error. There was no username provided.');
+    }
+    if (!password) {
+        throw new Error('Error. There was no password provided.');
+    }
+
     firstName = firstName || 'Anonymous';
     lastName = lastName || 'Incognito';
 
     createUser(userName, password, firstName, lastName);
 }
 
-function login(req, res) {
-    let users = getUsers();
+function login(userName, password) {
+    if (!userName) {
+        throw new Error('Error. There was no username provided.');
+    }
+    if (!password) {
+        throw new Error('Error. There was no password provided.');
+    }
 
-    const userName = req.body.userName;
-    const password = md5(req.body.password);
+    let users = getUsers();
 
     const user = users.find(
         (user) => user.userName === userName && user.password === password
     );
 
     if (!user) {
-        res.status(400).json({ status: 'No such user.' });
-        return;
+        throw new Error('Error. No such user.');
     }
 
     const sessionId = md5(v4()); //Should be REAL cryptography.
     user.session = sessionId;
-    setUsers(users);
-    res.cookie('userLoginSession', sessionId, {
-        sameSite: 'None',
-        secure: true,
-        httpOnly: true,
-        maxAge: 900000,
-    });
-    res.status(200).json({
-        status: 'OK',
-        username: user.userName,
-        statusMessage: 'You have successfully logged in.',
-    });
+    setUsers(user);
+
+    return sessionId;
 }
 
 function usersList() {
     return getUsers();
 }
 
-function logout(req, res) {
-    res.clearCookie('userLoginSession', {
-        sameSite: 'None',
-        secure: true,
-        httpOnly: true,
-        sameSite: 'None',
-    });
-    res.status(200).json({
-        status: 'OK',
-        statusMessage: 'You have successfully logged out.',
-    });
+function logout(userLoginSession) {
+    if (!userLoginSession) {
+        throw new Error('Error. There was no users session data provided.');
+    }
+
+    let users = getUsers();
+
+    const user = userLoginSession
+        ? users.find((user) => user.session === userLoginSession)
+        : null;
+    if (!user) {
+        throw new Error('You are not logged in.');
+    }
+    updateUser(user.id, 'session', 'null');
 }
 
 function updateUser(userId, key, value) {
+    if (!key) {
+        throw new Error("Error. You didn't provide any key to update.");
+    }
+    if (!value) {
+        throw new Error("Error. You didn't provide any value to updat.");
```

Comment:
Typo

Reply:
Fixed, will be commited with the next update.

Reply:
It's not fixed, unless it's committed.

Reply:
Commited. )